splinepy/microstructure/tiles/armadillo.py

```diff
@@ -74,4088 +85,5043 @@ def _closing_tile(
             )
 
-        v_wall_thickness = parameters[0, 0]
-        spline_list = []
-        v_zero = 0.0
-        v_one_half = 0.5
-        v_one = 1.0
-        v_half_contact_length = contact_length * 0.5
-        v_inner_half_contact_length = contact_length * parameters[0, 0]
-
-        if closure == "x_min":
-            # set points:
-            right = _np.array(
-                [
-                    [
-                        v_wall_thickness + v_one_half,
-                        -v_inner_half_contact_length + v_one_half,
-                        v_one_half + v_inner_half_contact_length,
-                    ],
-                    [
-                        v_one,
-                        -v_half_contact_length + v_one_half,
-                        v_one_half + v_half_contact_length,
-                    ],
-                    [
-                        v_wall_thickness + v_one_half,
-                        -v_inner_half_contact_length + v_one_half,
-                        v_one_half - v_inner_half_contact_length,
-                    ],
-                    [
-                        v_one,
-                        -v_half_contact_length + v_one_half,
-                        v_one_half - v_half_contact_length,
-                    ],
-                    [
-                        v_wall_thickness + v_one_half,
-                        v_inner_half_contact_length + v_one_half,
-                        v_one_half + v_inner_half_contact_length,
-                    ],
-                    [
-                        v_one,
-                        v_half_contact_length + v_one_half,
-                        v_one_half + v_half_contact_length,
-                    ],
-                    [
-                        v_wall_thickness + v_one_half,
-                        v_inner_half_contact_length + v_one_half,
-                        v_one_half - v_inner_half_contact_length,
-                    ],
-                    [
-                        v_one,
-                        v_half_contact_length + v_one_half,
-                        v_one_half - v_half_contact_length,
-                    ],
+        splines = []
+        for i_derivative in range(n_derivatives + 1):
+            if i_derivative == 0:
+                v_wall_thickness = parameters[0, 0]
+                v_zero = 0.0
+                v_one_half = 0.5
+                v_one = 1.0
+                v_half_contact_length = contact_length * 0.5
+                v_inner_half_contact_length = contact_length * v_wall_thickness
+            else:
+                v_wall_thickness = parameter_sensitivities[
+                    0, 0, i_derivative - 1
                 ]
-            )
+                v_zero = 0.0
+                v_one_half = 0.0
+                v_one = 0.0
+                v_half_contact_length = 0.0
+                v_inner_half_contact_length = contact_length * v_wall_thickness
```

Comment:
These are constants/parameters for a specific microtile. In the code below these declarations are the descriptions of the patches which the microstructure is made out of. In every of the patches' description these constants are in there (e.g. `v_inner_half_contact_length` is in the definition of the first patch). In the first loop iteration the geometry is constructed out of these patches. The loop iterations afterwards describe the analytical derivatives of each patch w.r.t. the `parameters`. That's why we have to hardcode these values, once for the geometry in the first loop iteration, and once for the derivatives in the subsequent iterations. That's also why we have to redefine `v_inner_half_contact_length` in the second loop iteration (in the else clause), because it depends on `v_wall_thickness` which is different in the if clause and the else clause. Each patch is defined once and hence we have to redefine these constants

Reply:
Otherwise we would have to define each patch a second time, once for the geometry and once for the derivatives. Then we could define two separate e.g. `v_inner_half_contact_length`s. But the additional definitions of these patches for the derivatives would add extra code and would lead to more difficulties in readability and modifiability